test/integration/runner.spec.js

```diff
@@ -0,0 +1,146 @@
+import { expect } from 'chai'
+import _ from 'lodash'
+
+import { ensureResources, createOrderByOrderNumber } from './test-utils.js'
+import { run } from '../../src/runner.js'
+import getLogger from '../../src/utils/logger.js'
+import { getApiRoot } from '../../src/utils/client.js'
+
+describe('runner', () => {
+  const apiRoot = getApiRoot()
+  const logger = getLogger()
+  let productIds
+  let shippingMethodId
+  const DAY_IN_MS = 24 * 60 * 60 * 1000
+
+  before(async () => {
+    const resources = await ensureResources(apiRoot, logger)
+    productIds = resources.productIds
+    shippingMethodId = resources.shippingMethodId
+  })
+
+  it('should create a template order from the checkout order', async () => {
+    const orderNumber = new Date().getTime().toString()
+    const checkoutOrder = await createOrderByOrderNumber(
+      apiRoot,
+      logger,
+      productIds,
+      shippingMethodId,
+      orderNumber
+    )
+    await run()
+
+    const {
+      body: { results: templateOrders },
+    } = await apiRoot
+      .orders()
+      .get({
+        queryArgs: {
+          where: `custom(fields(checkoutOrderRef(id="${checkoutOrder.id}")))`,
+          expand: 'paymentInfo.payments[*]',
+        },
+      })
+      .execute()
+
+    expect(templateOrders).to.have.lengthOf(4)
+    assertCustomFieldsEqual(checkoutOrder, templateOrders)
+    assertLineItemsEqual(checkoutOrder, templateOrders)
+    assertTemplateOrderCustomFields(checkoutOrder, templateOrders)
+    assertPayments(checkoutOrder, templateOrders)
+    const { body: checkoutOrderUpdated } = await apiRoot
+      .orders()
+      .withId({ ID: checkoutOrder.id })
+      .get()
+      .execute()
+    expect(checkoutOrderUpdated.custom.fields.isSubscriptionProcessed).to.be
```

Comment:
maybe we could also assert if the state=Active ?

Reply:
https://github.com/commercetools/commercetools-subscriptions/pull/4/commits/fefc278402c36902cdaf076c1ecb028293ae5c7f#diff-5c3fb411d68af19e71866dc939ebee5f3d60be0eac74b6f40064480dbb7db99dR74